core/src/main/java/com/uber/m3/tally/ScopeImpl.java

```diff
@@ -136,12 +144,32 @@ public void close() {
         }
     }
 
+    private Map<String, String> sanitizeTags(Map<String, String> tags) {
+        boolean hasChange = false;
+        for (Map.Entry<String, String> kv : tags.entrySet()) {
+            if (!sanitizer.sanitizeTagKey(kv.getKey()).equals(kv.getKey()) || !sanitizer.sanitizeTagValue(kv.getValue()).equals(kv.getValue())) {
+                hasChange = true;
+                break;
+            }
+        }
+        if (!hasChange) {
+            return tags;
+        }
+
+        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
+        if (tags != null) {
```

Comment:
nit: this check is redundant. If `tags` is null it will fail on `tags.entrySet()`

Reply:
I'd rather re-phrase this: @longquanzheng please move this check at the beginning to make sure there's no NPE

Reply:
yeah good point. Thanks